client/src/utils/responsiveness.ts

```diff
@@ -0,0 +1,54 @@
+import { useState, useEffect } from 'react'
+
+/**
+ * Given a CSS media query, returns whether the media query is matched, e.g.
+ *
+ * const isDesktopWidth = useMediaQuery('(min-width: 64em)')
+ *
+ * Listens for updates, e.g. in response to window resizing.
+ */
+export function useMediaQuery(query: string): boolean {
+  const [isMatched, setIsMatched] = useState(false)
+
+  useEffect(() => {
+    const matchMedia = window.matchMedia(query)
+    setIsMatched(matchMedia.matches)
+    const handleChange = () => {
+      setIsMatched(matchMedia.matches)
+    }
+    matchMedia.addEventListener('change', handleChange)
+    return () => {
+      matchMedia.removeEventListener('change', handleChange)
+    }
+  }, [query])
+
+  return isMatched
+}
+
+/** A breakpoint for mobile vs. tablet */
+export const BREAKPOINT_M = '40em'
+/** A breakpoint for tablet vs. desktop */
+export const BREAKPOINT_L = '64em'
```

Comment:
Oh the art of picking breakpoints. I consulted a few articles (there are hundreds). Here's one that I took inspiration from: https://polypane.app/blog/the-breakpoints-we-tested-in-2021-and-the-ones-to-test-in-2022

Reply:
Yes good q! This article walks through the reasoning for using `em` over `rem` or `px`: https://zellwk.com/blog/media-query-units/ (some of it involves the quirks/limitations of Safari in particular)